front/src/applications/stdcm/views/StdcmView.tsx

```diff
@@ -237,4 +242,42 @@ const StdcmView = () => {
   );
 };
 
+const StdcmView = () => {
+  const [isDebugMode, setIsDebugMode] = useState(false);
+  const { loading, error, loadStdcmEnvironment } = useStdcmEnvironment();
+  const stdcmConf = useSelector(getStdcmConf);
+
+  const isStdcmConfValid = useMemo(
+    () => !!stdcmConf.searchDatetimeWindow && !!stdcmConf.timetableID && !!stdcmConf.infraID,
+    [stdcmConf]
+  );
+
+  useEffect(() => {
+    if (!isDebugMode) {
+      loadStdcmEnvironment();
+    }
+  }, [isDebugMode]);
+
+  // If we've got an error during the loading of the stdcm env which is not the "no config error" message,
+  // we let the error boundary manage it
+  if (error && error.message !== NO_CONFIG_FOUND_MSG) throw error;
+
+  // When the STDCM environment is being loaded and the conf is not valid, we do not mount the actual STDCM
+  // view content yet:
+  if (loading && !isStdcmConfValid) return null;
+
+  if (!isStdcmConfValid) {
+    throw new Error('STDCM conf is not valid.');
+  }
```

Comment:
With this check, if we have no config, then I believe that the stdcmConf will not be valid right ? So will we display the `StdcmEmptyConfigError` ?

Reply:
Actually, now, if I remove that check and at some point the STDCM conf is not valid, then the call to `useStdcm` from `StdcmViewContent` will probably fail at some point. For instance, [this line](https://github.com/OpenRailAssociation/osrd/blob/3140e65a493f3d753931fbc512834be93523b288/front/src/applications/stdcm/hooks/useStdcm.ts#L56) will throw a `Value timetableID of RootState is undefined, while the selector should remain non-nullable` error (now thrown by [`makeSubSelector`](https://github.com/OpenRailAssociation/osrd/blob/41fa8386154fb31939c4f0162ee831f9b3249dc3/front/src/utils/selectors.ts#L26-L28)).